data/types/base/pump/settings.go

```diff
@@ -87,6 +85,17 @@ func (s *Settings) Validate(validator data.Validator) {
 		}
 	}
 
+	if s.BasalSchedules != nil {
```

Comment:
You need to add an additional level of validator since this appears to be an array WITHIN an array.  So:
```
	if s.BasalSchedules != nil {
		basalSchedulesValidator := validator.NewChildValidator("basalSchedules")
		for basalScheduleName, basalSchedule := range *s.BasalSchedules {
			// TODO: Do we want to validate the basalScheduleName?  Length > 0?
			if basalSchedule != nil {
				basalScheduleValidator := basalSchedulesValidator.NewChildValidator(basalScheduleName)
				for basalScheduleItemIndex, basalScheduleItem := range *basalSchedule {
					basalScheduleItem.Validate(basalSchedulesValidator.NewChildValidator(basalScheduleItemIndex))
				}
			}
		}
	}
```